File: android/app/src/main/java/com/nomenas/wrapperconcept/project/SingleInheritance.java
package com.nomenas.wrapperconcept.project;

/**
 * Created by naum.puroski on 13/04/2017.
 */

public class SingleInheritance extends Object implements SimpleInterface {
    public SingleInheritance() {create();};

    public final native void method1();
    public final native int method2(int value);

    public native int java_overridable();

    protected native void create();
    protected native void delete();
}

File: project/SingleInheritance.cpp
#include "SingleInheritance.h"


void SingleInheritance::method1() const {
    ++_counter;
}

int SingleInheritance::method2(int value) {
    return ++_counter + value;
}

int SingleInheritance::java_overridable() {
    return 0;
}

int SingleInheritanceExt::method2(int value) {
    return ++_counter + (value * 2);
}